src/util/kaldi-table-inl.h

```diff
@@ -340,6 +441,13 @@ template<class Holder>  class SequentialTableReaderScriptImpl:
     // or the user called FreeCurrent().. note,
     // if when called by user we are in this state,
     // it means the user called FreeCurrent().
+    kObjectLoaded, // Called NextScpLine() and found we already  yes         yes
```

Comment:
There is already a member variable named data_rxfilename_, right?  Can't
you always ensure that if that variable is set and the state if one of the
states that indicates we have an object, then the holder_ contains the
object it points to?  That way when you do NextScpLine() or whatever it's
called, you can decide whether to set the state to kHaveScpLine, or
kLoadSucceeded, depending on whether the new rxfilename is the same as the
old.

Dan


On Thu, Mar 24, 2016 at 10:05 PM, Xiaohui Zhang <notifications@github.com>
wrote:

> In src/util/kaldi-table-inl.h
> <https://github.com/kaldi-asr/kaldi/pull/611#discussion_r57412771>:
>
> > @@ -340,6 +441,13 @@ template<class Holder>  class SequentialTableReaderScriptImpl:
> >      // or the user called FreeCurrent().. note,
> >      // if when called by user we are in this state,
> >      // it means the user called FreeCurrent().
> > +    kObjectLoaded, // Called NextScpLine() and found we already  yes         yes
>
> Since there is no function to test whether holder_ is empty, we need
> either a string like data_rxfilename_last_ (which I used before) or an
> extra state like this (which you asked me to add) to tell LoadCurrent
> whether to load the object from disk.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/611/files/d2c8803477273b8c96d2aacebdba9f93ea780781#r57412771>
>